R/interpret.R

```diff
@@ -61,8 +62,36 @@ interpret <- function(x, sparsity = NULL, remove_zeros = TRUE, top_n) {
     coefs <- coefs[, j, drop = FALSE]
     lambda <- g$lambda[j]
   }
+
+  ref_levels <- x %>%
+    attr("recipe") %>%
+    attr("ref_levels")
+
   coefs <-
-    tibble::tibble(variable = rownames(coefs), coefficient = coefs[, 1]) %>%
+    tibble::tibble(variable = rownames(coefs), coefficient = coefs[, 1],
+                   reference_level = map_chr(variable, ~{
```

Comment:
This code needs to be refactored for a number of reasons. I've asked you to remove it from the data frame, so it needs to go somewhere else, but it also strikes me as quite complicated for what it's doing. And complication is the enemy, because it makes code harder to understand, easier to be buggy, and harder to change later. Would it maybe make things easier to store the factor reference levels with the factor levels (in the recipe attributes) to be able to match non-reference levels to their reference levels? Maybe create that lookup table in prep and carry it along as a recipe attr? I'm not sure, but you get to create the objects (in prep data) that you're working with here, so you should be able to simplify.

Also, I'm not sure if this is spelled out in our contributing guide (it should be; my bad if it's not), but never program with `sapply`. It is not type stable: it could return a list, vector, or matrix depending on what comes back, and that's asking for trouble. Use `purrr::map_XXX` instead. Likewise, `stringr::` functions are preferred to base R regex functions.

Reply:
I remember vaguely something about not using `sapply`. I will not try to use it again. Thanks for the ideas. Hopefully you like my new implementation.